lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+
+'''
+    LCLS Naming Tool
+
+    A tool for checking the form and content (but not meaning) of PV names with
+    respect to the LCLS naming convention.
+
+    A valid PV name is in the format FFFFF:GGG:CCC:NN:XXXX, where GGG and NN are
+    optional and can be omitted.
+
+    This script only checks the validity of PV names (not devices).
+
+    Instructions:
+    1. Set $ chmod 775 lcls_naming_tool.py to make the file executable.
+
+    2. To check if a PV name is valid type pipe in the text like so 
+            $ echo "XCS:DG3:GCC:02:PCTRLSPRBCK" | ./lcls_naming_tool.py
+    
+    3. To check if a list of PV names is valid pipe in the file name like so 
+            $ cat pvlist.txt | ./lcls_naming_tool.py
+'''
+
+
+import sys
+import json
+import re
+
+
+fc_valid = True
+fg_valid = True
+ccc_valid = True
+nn_valid = True
+cspv_valid = True
+fc_dict = {}
+fg_dict = {}
+ccc_dict = {}
+beam_sources = {'K': '', 'L': ''}
+beam_numbers = {'0': '', '1': '', '2': '', '3': '', '4': '', '5': ''}
+
+
+def load_taxons():
+    global fc_dict
+    global fg_dict
+    global ccc_dict
+
+    # Read the json files containing all the taxons
+    with open('functional_component_taxon.json') as fc_file:
+        fc_dict = json.load(fc_file)
+
+    with open('fungible_element_taxon.json') as fg_file:
+        fg_dict = json.load(fg_file)
+
+    with open('ccc_taxon.json') as ccc_file:
+        ccc_dict = json.load(ccc_file)
+
+
+def functional_component_is_valid(fc_taxon):
+    
+    # check length of the functional component
+    if (len(fc_taxon) == 5):
+        fc_prefix = str(fc_taxon[0] + fc_taxon[1])
```

Comment:
This is a good place to use the str slice syntax
```suggestion
        fc_prefix = fc_taxon[:2]
```
edit from future me: I see that the code was originally structured for the input to be a list of strings